velox/dwio/parquet/reader/ParquetReader.cpp

```diff
@@ -567,24 +567,16 @@ void ReaderBase::scheduleRowGroups(
     const std::vector<uint32_t>& rowGroupIds,
     int32_t currentGroup,
     StructColumnReader& reader) {
-  auto thisGroup = rowGroupIds[currentGroup];
-  auto nextGroup =
-      currentGroup + 1 < rowGroupIds.size() ? rowGroupIds[currentGroup + 1] : 0;
-  auto input = inputs_[thisGroup].get();
-  if (!input) {
-    inputs_[thisGroup] = reader.loadRowGroup(thisGroup, input_);
-  }
-  for (auto counter = 0; counter < options_.prefetchRowGroups(); ++counter) {
-    if (nextGroup) {
-      if (inputs_.count(nextGroup) == 0) {
-        inputs_[nextGroup] = reader.loadRowGroup(nextGroup, input_);
-      }
-    } else {
-      break;
+  auto numRowGroupsToLoad = std::min(
+      options_.prefetchRowGroups() + 1,
+      static_cast<int64_t>(rowGroupIds.size() - currentGroup));
+  for (auto i = 0; i < numRowGroupsToLoad; i++) {
+    auto thisGroup = rowGroupIds[currentGroup + i];
+    if (!inputs_[thisGroup]) {
+      inputs_[thisGroup] = reader.loadRowGroup(thisGroup, input_);
     }
```

Comment:
else break to match the old behavior?  Not sure whether this is actually happening though

Reply:
I think we won't need the break since we compute the number of rowGroups to prefetch before the loop.

Reply:
> else break to match the old behavior? Not sure whether this is actually happening though

That's exactly the point of this change: we calculate the number of row groups to load before entering the loop, so there is no need to do the "else break" check in every iteration. numRowGroupsToLoad is guarded by the total number of rowgroups and will not go out of bound.